src/AccessControl/SimpleObjectPolicies.py

```diff
@@ -119,3 +120,13 @@ def allow_type(Type, allowed=1):
     if has_values:
         assert key_type is type(tree.values())
         assert key_type is type(tree.items())
+
+
+@contextmanager
+def override_containers(type_, assertions):
+    """Temporarily override the container assertions."""
+    orig_container = Containers(type_)
+    ContainerAssertions[type_] = assertions
+    yield
+    if orig_container is not None:
+        ContainerAssertions[type_] = orig_container
```

Comment:
If `orig_container` is `None`, shouldn't `ContainerAssertions[type_]` be deleted?